repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -236,10 +261,46 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
 
     if (Option(configuration).map(_.getPolicyAcquisition.getCache.getTtl) != Option(newConfiguration.getPolicyAcquisition.getCache.getTtl)) {
       cache = CacheBuilder.newBuilder()
-                          .expireAfterWrite(newConfiguration.getPolicyAcquisition.getCache.getTtl, TimeUnit.SECONDS)
-                          .build(new CacheLoader[String, XsltExecutable]() {
-                            override def load(key: String): XsltExecutable = getPolicy(key)
-                          })
+        .expireAfterWrite(newConfiguration.getPolicyAcquisition.getCache.getTtl, TimeUnit.SECONDS)
+        .build(new CacheLoader[String, XsltExecutable]() {
+          override def load(key: String): XsltExecutable = getPolicy(key)
+        })
+    }
+
+    try {
+      // Create a DOM XMLSignatureFactory that will be used to
+      // generate the enveloped signature.
+      fac = XMLSignatureFactory.getInstance("DOM")
+      // Create a Reference to the enveloped document (in this case,
+      // you are signing the whole document, so a URI of "" signifies
+      // that, and also specify the SHA1 digest algorithm and
+      // the ENVELOPED Transform.
+      val ref = fac.newReference(
+        "",
+        fac.newDigestMethod(DigestMethod.SHA1, null),
+        Collections.singletonList(fac.newTransform(Transform.ENVELOPED, null.asInstanceOf[TransformParameterSpec])),
+        null,
+        null)
+      // Create the SignedInfo.
+      si = fac.newSignedInfo(fac.newCanonicalizationMethod(
+        CanonicalizationMethod.INCLUSIVE,
+        null.asInstanceOf[C14NMethodParameterSpec]),
+        fac.newSignatureMethod(SignatureMethod.RSA_SHA1, null),
+        Collections.singletonList(ref))
+      // Load the KeyStore and get the signing key and certificate.
+      val creds = newConfiguration.getSignatureCredentials
+      val keyStoreFilename = FileUtilities.guardedAbsoluteFile(configRoot, creds.getKeystoreFilename).getAbsolutePath
```

Comment:
Yes, we might want to do that. Then we would have to open the same keystore multiple times either.